libsql-replication/src/snapshot.rs

```diff
@@ -60,6 +66,9 @@ impl SnapshotFile {
                 let mut frame: MaybeUninit<FrameBorrowed> = MaybeUninit::uninit();
                 let buf = unsafe { std::slice::from_raw_parts_mut(frame.as_mut_ptr() as *mut u8, size_of::<FrameBorrowed>()) };
                 self.file.read_exact(buf).await?;
+                if let Some(encryptor) = &self.encryptor {
+                    encryptor.decrypt(&mut buf[size_of::<FrameHeader>()..]).map_err(|_| Error::InvalidSnapshot)?;
+                }
```

Comment:
can we move that bit after `assume_init`, and use `frame.page_mut()`

Reply:
yessir, done